src/core/config/transformer/improve_type_names/improve_type_names.rs

```diff
@@ -114,22 +115,47 @@ impl<'a> CandidateGeneration<'a> {
 pub struct ImproveTypeNames;
 
 impl ImproveTypeNames {
-    /// Generates type names based on inferred candidates from the provided
-    /// configuration.
     fn generate_type_names(&self, mut config: Config) -> Config {
+        let ai_generated_type_names = ImproveTypeNamesLLM::generate_llm_type_names(config.clone());
         let finalized_candidates = CandidateGeneration::new(&config).generate().converge();
+        let mut llm_worked = false;
 
         for (old_type_name, new_type_name) in finalized_candidates {
+            // Try to use the llm generated type names first
             if let Some(type_) = config.types.remove(old_type_name.as_str()) {
-                // Add newly generated type.
-                config.types.insert(new_type_name.to_owned(), type_);
-
-                // Replace all the instances of old name in config.
-                for actual_type in config.types.values_mut() {
-                    for actual_field in actual_type.fields.values_mut() {
-                        if actual_field.type_of == old_type_name {
-                            // Update the field's type with the new name
-                            actual_field.type_of.clone_from(&new_type_name);
+                if let Ok(ai_generated_types) = &ai_generated_type_names {
+                    if let Some(ai_generated_types) = ai_generated_types.get(&old_type_name) {
+                        // ensure the type name is not already present in the config, else try
+                        // the next llm generated type name
+                        for type_name in ai_generated_types.iter() {
+                            if !config.types.contains_key(type_name) {
+                                llm_worked = true;
+                                config.types.insert(type_name.to_owned(), type_.clone());
+                                // Replace all the instances of old name in config.
+                                for actual_type in config.types.values_mut() {
+                                    for actual_field in actual_type.fields.values_mut() {
+                                        if actual_field.type_of == old_type_name {
+                                            // Update the field's type with the new name
+                                            actual_field.type_of.clone_from(type_name);
+                                        }
```

Comment:
```rust
input PostInput {
    ...
}

post(postInput: PostInput): Post
```

we use AI to generate the name of PostInput, then we'll end up renaming the input type but not in the args. so should also iterate over args present in the field.

Reply:
Sure will try. But in this PR, AI is only updating the types that the inference would have updated.